File: 0696-count-binary-substrings/0696-count-binary-substrings.py
class Solution:
    def countBinarySubstrings(self, s: str) -> int:
        i = 0
        ans = prev = 0
        n = len(s)
        
        while(i < n):
            curr = 1
            while(i < n - 1 and s[i] == s[i + 1]):
                i += 1
                curr += 1
            
            ans += min(prev, curr)
            prev = curr
            i += 1
        
        return ans
                

File: 0696-count-binary-substrings/0696-count-binary-substrings.go
func countBinarySubstrings(s string) int {
    prev, streak, res := 0, 1, 0
    for i := range len(s) - 1 {
        if s[i] == s[i + 1] {
            streak++
        } else {
            prev = streak
            streak = 1
        }

        if streak <= prev{
            res++
        }
    }

    return res
}